activation/post_verifier_test.go

```diff
@@ -100,3 +101,16 @@ func TestPostVerifierClose(t *testing.T) {
 	err := v.Verify(context.Background(), &shared.Proof{}, &shared.ProofMetadata{})
 	require.EqualError(t, err, "verifier is closed")
 }
+
+func TestPostVerifierPrioritization(t *testing.T) {
```

Comment:
This test doesn't actually check whether the specified `nodeID` takes the priority, right?
The actual check for the ordering would probably involve
1) Starting a verification that blocks (e.g. wait for a channel to be closed)
2) While it's blocked, request some more verifications, some prioritized and some not. By having more than 1 of each type, it's possible to get higher chances of test failing due to bad order
3) Unblock the initial verification
4) Wait for all the verifications to finish
5) Verify the order in which they were executed (prioritized must go first)

To check that the test verifies thing properly, you can collapse nested `select`s into 1 and then the test should fail, at least in most runs

I'm not insisting on these changes as they may be a bit too involved for such a simple thing, but just as some thoughts...

Reply:
Yeah, you are right, it merely checks if the code path "works". I had a test similar to what you described but then changed the implementation to use 2 channels and couldn't figure out how to make a test for the new approach. Thanks for your suggestion - I will try it out :)

Reply:
We agreed offline that it's too much hassle (it would require somehow refactoring the code to add a sync-point for the unit test only).